src/term/mod.rs

```diff
@@ -453,13 +453,28 @@ impl<'a> Iterator for RenderableCellsIter<'a> {
                 (cell, selected)
             };
 
-            // Apply inversion and lookup RGB values
-            let mut fg_rgb = self.compute_fg_rgb(cell.fg, &cell);
-            let mut bg_rgb = self.compute_bg_rgb(cell.bg);
+            // If selection color wasn't set by user we should invert cell colors.
+            let selection_background_color = self.config.selection_background_color();
+            let invert_selection = selected && selection_background_color.is_none();
+
+            // Set custom selection color and selection text color, if they were provided.
+            let (bg_color, fg_color) = if selected {
+                (selection_background_color.unwrap_or_else(|| cell.bg ),
+                    self.config.selection_text_color().unwrap_or_else(|| cell.fg))
```

Comment:
The formatting here is a bit messy, I propose doing it like this:

```rust
(
    selection_background_color.unwrap_or_else(|| cell.bg),
    self.config.selection_text_color().unwrap_or_else(|| cell.fg),
)
```

This should probably feel a bit more uniform and I think it's probably more like what rustfmt would do by itself.